test/tests/tagsboxTest.js

```diff
@@ -16,15 +16,18 @@ describe("Item Tags Box", function () {
 	
 	
 	describe("Tag Editing", function () {
-		it.skip("should update tag when pressing Enter in textbox", async function () {
+		it("should update tag when pressing Enter in textbox", async function () {
 			// editable-text behavior relies on focus, so we first need to bring the window to the front.
 			let win = Zotero.getMainWindow();
 			let activatePromise = new Promise(
 				resolve => win.addEventListener('activate', resolve, { once: true })
 			);
 			Zotero.Utilities.Internal.activate();
 			Zotero.Utilities.Internal.activate(win);
-			await activatePromise;
+			// Required on macOS but causes a timeout on linux
+			if (!Zotero.isLinux) {
+				await activatePromise;
+			}
```

Comment:
If we're not going to await it we don't need to add the event listener above either.

And is it really that waiting for activation isn't required on Linux, or that activation isn't working, or that the Z window is automatically in front on Linux, or…? And what's the actual behavior this is relying on?

Reply:
Ok, yeah, what I re-pushed should make more sense now. As I understand, the window is already at the front on Linux, so all activation should just be skipped. I swapped the condition from checking the platform to `doc.hasFocus()` because I think it better reflects when the activation should happen